src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Window.cs

```diff
@@ -2511,26 +2515,21 @@ internal void CreateSourceWindow(bool duringShow)
                 UnsafeNativeMethods.ChangeWindowMessageFilterEx(_swh.CriticalHandle, WindowMessage.WM_COMMAND, MSGFLT.ALLOW, out info);
             }
 
-            if (Standard.Utility.IsOSWindows11OrNewer && ThemeColorization.IsFluentWindowsThemeEnabled)
+            if (Standard.Utility.IsOSWindows11OrNewer && ThemeManager.IsFluentWindowsThemeEnabled)
             {
-                ResourceDictionary themeDictionary = new ResourceDictionary();
+                var themeColorFileName = ThemeManager.GetUseLightTheme() ? "light.xaml" : "dark.xaml";
+                ResourceDictionary themeDictionary = new ResourceDictionary() 
+                    { Source = new Uri($"pack://application:,,,/PresentationFramework.FluentWindows;component/resources/theme/{themeColorFileName}", 
+                                            UriKind.Absolute) };
 
-                if(ThemeColorization.IsThemeDark())
-                {
-                    themeDictionary.Source = new Uri("pack://application:,,,/PresentationFramework.FluentWindows;component/resources/theme/dark.xaml", UriKind.Absolute);
-                }
-                else 
-                {
-                    themeDictionary.Source = new Uri("pack://application:,,,/PresentationFramework.FluentWindows;component/resources/theme/light.xaml", UriKind.Absolute);
-                }
-
-                ThemeColorization.AddThemeDictionary(themeDictionary);
+                ThemeManager.AddFluentWindowsThemeDictionary(themeDictionary);
```

Comment:
You use ThemeManager to figure out the dictionary, why not let it to the ThemeManager? (i.e. make it a parameterless method)

Reply:
Yeah, working on that. I am changing it to an initialization based pattern. i.e.

From window, we will call InitializeFluentWindowsTheme and ApplySystemTheme

And the initialization method will take care of figuring out the resource dictionary.